nautobot/ipam/models.py

```diff
@@ -1120,6 +1108,7 @@ def clean(self):
                     }
                 )
             self.parent = closest_parent
+            self._namespace = None
```

Comment:
Shouldn't this be updating to `self.parent.namespace` instead of `None`? Same question below.

(edit) Oh, this is because setting `self._namespace = None` actually means "use self.parent.namespace"? That feels pretty unintuitive to me.

Reply:
Actually, I don't think we should set `self._namespace = self.parent.namespace` because the parent's namespace can be changed. If we cache the namespace and then the parent's namespace changes, our cache is incorrect.

Reply:
If we implement the `parent.setter` this will probably be easier to understand

Reply:
It may also be better to move that logic out of `self._namespace` and into the places where it's actually used.